relayer/relayer/src/contracts/azero.rs

```diff
@@ -97,6 +262,48 @@ impl MostInstance {
         })
     }
 
+    pub async fn get_base_fee(&self, connection: &Connection) -> Result<u128, AzeroContractError> {
+        Ok(self
+            .contract
+            .read0::<Result<u128, _>, _>(connection, "get_base_fee", Default::default())
+            .await??)
+    }
+
+    pub async fn send_request(
+        &self,
+        signed_connection: &AzeroConnectionWithSigner,
+        src_token_address: [u8; 32],
+        amount: u128,
+        dest_receiver_address: [u8; 32],
+        base_fee: u128,
+    ) -> Result<TxInfo, AzeroContractError> {
+        let gas_limit = Weight {
+            ref_time: self.ref_time_limit,
+            proof_size: self.proof_size_limit,
+        };
+        let args = [
+            bytes32_to_str(&src_token_address),
+            amount.to_string(),
+            bytes32_to_str(&dest_receiver_address),
+        ];
+
+        let params = ExecCallParams::new().gas_limit(gas_limit).value(base_fee);
+
+        // Exec does dry run first, so there's no need to repeat it here
+        self.contract
+            .exec(signed_connection, "send_request", &args, params)
+            .await
+            .map_err(|why| {
+                error!("send_request failure: {why:?}");
+                AzeroContractError::SendRequestTxFailure {
```

Comment:
I think that mapping onto a generic error here is not a good idea, made sense when we didn't dry run before execution because of inscrutable errors - now revert errors should be readable